docs/DeveloperGuide.md

```diff
@@ -1,38 +1,536 @@
+information need to be extracted from arguments in the user input.
+
+It simply breaks up an entire ```/argument [payload]``` String into ```/argument``` and
+```[payload]``` and returns the `[payload]`. Hence, any String passed into this method must
+take on the form of ```/argument [payload]``` or an exception will be thrown and handled by
+printing an error message to the CLI.
+
+`extractQuestionDifficulty`
+
+This method acts as a helper method for extracting an `QnDifficulty` enum type from a user
+text input. This is used for setting question difficulty during the construction of a
+`Question` , or the execution of a `CommandMarkDifficulty`.
+
+If the string passed to this method does not describe an existing question difficulty level, an
+exception will be thrown and handled by printing an error message to the CLI.
+
+## Command Components
+
+### Help Command - Display Commands
+
+When executed, this command will execute a standard Final String containing all 
+the commands available for the user to use, as well as the format they are meant
+to be written in. 
+
+This same String is also displayed when an invalid command is used.
+
+### Short Command - Add Short Answer Question to the Quiz
+
+One of the supported question formats is the Short Answer Question, in which a question
+can be answered with a phrase of a few words, which will be matched to a corresponding answer
+of sufficient similarity.
+
+It is parsed as: `short [question]/[answer]/[module]/[difficulty]`
+1. `[question]` is the question, phrased as asking the user (i.e. What is 2 + 2? )
+2. `[answer]` is the answer or possible answer(s) that the user must
+   input to count as correct. It is case-insensitive. (i.e. 4, four)
+3. `[module]` is the module that the question belongs in (i.e. CS2113)
+4. `[difficulty]` is the difficulty of the question for sorting later (i.e. Hard)
+
+*Condensed Class Diagram - Does not contain all attributes & methods
+![](./UML/AddShortCommand.jpg)
+
+Thereafter, the command is returned to the Quizhub component and executed,
+to add a Question object to the corresponding QuestionList object
+using the `addToQuestionList` method. This method will analyse the arguments above
+and add the Question to the appropriate list.
+
+![](./UML/Commands/Initial State.png)
+![](./UML/Commands/commandAddState.png)
+
+### List Command - Show all Questions with Index
+
+Lists all the questions in the current QuestionList. 
+
+When executed, the command will invoke the QuestionList.printQuestionList method. 
+This method first checks whether the list is empty and prints and error message if empty, 
+else it will invoke the QuestionList.printQuestion method on each Question object, 
+with the asList parameter set as true. The QuestionList.printQuestion method will then
+print each question with a given index, and indicate the question's type and completion status
+(obtained through Question.getQuestionType() and Question.questionIsDone() methods)
```

Comment:
I think this can be formatted better, eg. using `QuestionList.printQuestionList`